components/net/tests/main.rs

```diff
@@ -14,6 +14,7 @@ mod filemanager_thread;
 mod hsts;
 mod http_cache;
 mod http_loader;
+mod local_index;
```

Comment:
I agree, but having both the library code file and the unit test file with the exact same name led to an hour of fighting with the compiler to get it to refer to the intended thing, so I changed the name to make it easier to target the two different files.

Once the shape of these new files settles down (sounds like a lot more changes are needed) then we can rename this unit test file so that its name exactly matches the library code file.

Reply:
> Once the shape of these new files settles down (sounds like a lot more changes are needed)

I don't think any changes to the file structure, other than that suggested here, is necessary. 

> I agree, but having both the library code file and the unit test file with the exact same name led to an hour of fighting with the LSP/editor

If you look at the other files in the test folder, they all have the same name as the folder they are testing. I don't know about the problem you encountered, but I don't think it is a reason to move away from a clearly established pattern in the project.

Reply:
No, it was only temporary (and has now been successfully renamed on my local machine). Having two files with the same name in the same package was just causing the LSP to suggest the wrong one over and over again, leading to compilation problems. Giving them separate names just meant that it became clear which one the LSP was trying to feed me, so that I could avoid mix-ups and get everything to compile.